### DISCUSSION TITLE
Single line if (without else) statement

### DISCUSSION THREAD

**Discussion by Cationiz3r** [rust-lang/rustfmt#6369] [Q&A]
Hello all, I was working on a project of mine when I found that `rustfmt` was always expanding short `if` statements.

For example, I have a function that adds cached items into a redis server, which would always fail if the items list is empty, so the following code snippet is used:
```rust
if items.is_empty() { return Ok(()); }
```

However, by default, `rustfmt` always expand it to:
``` rust
if items.is_empty() {
    return Ok(());
}
```
which takes 2 extra lines.

Is there an option I can use to compact it into the first snippet?

**Comment by wellkilo** [comment]
There is no rustfmt option that keeps a standalone `if` statement such as this on one line:

```rust
if items.is_empty() { return Ok(()); }
```

The similarly named stable option `single_line_if_else_max_width` does **not** cover this case. It applies when an `if ... else ...` is formatted as a subexpression. In the current formatter, `ControlFlow::new_if` sets `allow_single_line` only when `expr_type == ExprType::SubExpression`, and `rewrite_single_line` immediately requires an `else_block`. See [`to_control_flow`](https://github.com/rust-lang/rustfmt/blob/9e92f7b48cf49412c647eb6b20b6e4315481a23f/src/expr.rs#L714-L731) and [`rewrite_single_line`](https://github.com/rust-lang/rustfmt/blob/9e92f7b48cf49412c647eb6b20b6e4315481a23f/src/expr.rs#L870-L892).

So the idiomatic, rustfmt-stable form is the expanded block. If you must preserve the compact guard locally, the available escape hatch is:

```rust
#[rustfmt::skip]
fn add_cached(items: &[Item]) -> Result<(), Error> {
    if items.is_empty() { return Ok(()); }
    // ...
}
```

but `#[rustfmt::skip]` applies to the containing item, not just that one `if`, so it is usually a worse trade-off than accepting the standard layout. `single_line_if_else_max_width` cannot be used to change this specific statement form.

If this answers which setting applies, please mark the reply as accepted.
